TestHelper.psm1

```diff
@@ -201,3 +201,216 @@ function Install-ResourceDesigner {
     
     return (Get-Module -Name $Script:DesignerModuleName -ListAvailable)
 }
+
+
+<#
+    .SYNOPSIS
+        Initializes an enviroment for running unit or integration tests
+        on a DSC resource.
+        
+        This includes the following things:
+        1. Backing up any modules matching the name of the module being tested.
+        2. Copying this module to the PSModules folder.
+        3. Backing up any settings that need to be changed to accurately test
+           a resource.
+        4. Producing a test object containing any parameters that may be used
+           for testing as well as storing the backed up settings.
+    
+    .PARAMETER DSCModuleName
+        The name of the DSC Module containing the resource that the tests will be
+        run on.
+
+    .PARAMETER DSCResourceName
+        The full name of the DSC resource that the tests will be run on. This is
+        usually the name of the folder containing the actual resource MOF file.
+
+    .PARAMETER TestType
+        Specifies the type of tests that are being intialized. It can be:
+        Unit: Initialize for running Unit tests on a DSC resource. Default.
+        Integration: Initialize for running Integration tests on a DSC resource.
+
+    .OUTPUT
+        Returns a test environment object which must be passed to the
+        Restore-TestEnvironment function to allow it to restore the system
+        back to the original state as well as clean up and working/temp files.
+        
+    .EXAMPLE
+        $TestEnvrionment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            
+        This command will initialize the test enviroment for Unit testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.      
+
+    .EXAMPLE
+        $TestEnvrionment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            -TestType Integration
+            
+        This command will initialize the test enviroment for Integration testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.    
+#>
+function Initialize-TestEnvironment {
+    [OutputType([PSObject])]
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCModuleName,
+        
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCResourceName,
+
+        [ValidateSet('Unit','Integration')]
+        $TestType = 'Unit'
```

Comment:
Should probably be [String] $TestType to match the others.

Also do we want to set Unit as the default? Ideally we should be checking in with both unit/integration tests so we do want to be giving priority to one?